Add None.toString

bmaslakov opened the pull request:
I often use this class for debugging, and calling something like `println("sessionId=$optional")` gives this:
```
sessionId=com.gojuno.koptional.None@7ce026d3
```
Why don't we override `toString`? :)
```
sessionId=None
```

bmaslakov commented:
As a side note, it'd be also great to have `Some(123)` instead of `Some(value=123)`

artem-zinnatullin commented on koptional/src/main/kotlin/com/gojuno/koptional/Optional.kt:
```diff
@@ -9,6 +9,8 @@ sealed class Optional<out T : Any> {
 
 data class Some<out T : Any>(val value: T) : Optional<T>()
-object None : Optional<Nothing>()
+object None : Optional<Nothing>() {
```
nit: maybe add an empty line between classes now? 

With additional code it now looks a little bit crumpled :)

artem-zinnatullin commented:
I guess the only use case where `com.gojuno.koptional.None@7ce026d3` is useful is dealing with scenarios like https://youtrack.jetbrains.com/issue/KT-14540 when you accidentally end up having multiple instances of `object`

But I'd still merge this PR :)

bmaslakov commented:
Fixed the line separator and added a cleaner version of `Some.toString`.
As for KT-14540, I think the proper workaround would be to add `None.equals(other) = other is None`, but I think it is really out of scope for this PR.

artem-zinnatullin commented on koptional/src/main/kotlin/com/gojuno/koptional/Optional.kt:
```diff
@@ -8,7 +8,12 @@ sealed class Optional<out T : Any> {
-data class Some<out T : Any>(val value: T) : Optional<T>()
-object None : Optional<Nothing>()
+data class Some<out T : Any>(val value: T) : Optional<T>() {
+    override fun toString() = "Some($value)"
```
Let's add simple unit tests btw?

artem-zinnatullin commented on koptional/src/main/kotlin/com/gojuno/koptional/Optional.kt:
```diff
@@ -8,7 +8,12 @@ sealed class Optional<out T : Any> {
+}
+
+object None : Optional<Nothing>() {
+    override fun toString() = "None"
```
And here

bmaslakov commented:
Did my best :)

artem-zinnatullin commented on koptional/src/test/kotlin/com/gojuno/koptional/OptionalSpec.kt:
```diff
@@ -49,4 +49,38 @@ class OptionalSpec : Spek({
+
+        }
+
+        context("Some<Object>.toString") {
```
This one seems unnecessary, but we can leave it :)

artem-zinnatullin commented:
@ming13 @nostra13 PTAL

artem-zinnatullin commented:
MAYBE WE EVEN NEED TO BUMP MAJOR VERSION
